packages/backend-function/src/resolve_secret_banner.ts

```diff
@@ -0,0 +1,31 @@
+import { SSM } from '@aws-sdk/client-ssm';
+
+/**
+ * The body of this function will be used to resolve secrets for Lambda functions
+ */
+export const resolveSecretBanner = async (client?: SSM) => {
+  if (!client) {
+    client = new SSM();
+  }
+  const envPathObj: Record<string, string> = JSON.parse(
+    process.env.AMPLIFY_SECRET_PATHS ?? '{}'
+  );
+  const paths = Object.values(envPathObj);
+  if (paths.length > 0) {
+    const response = await client.getParameters({
+      Names: paths,
+      WithDecryption: true,
+    });
+    if (response.Parameters && response.Parameters?.length > 0) {
+      for (const parameter of response.Parameters) {
+        for (const [env, path] of Object.entries(envPathObj)) {
+          if (parameter.Name === path) {
+            process.env[env] = parameter.Value;
+          }
+        }
+      }
+    }
+  }
+};
+
+await resolveSecretBanner();
```

Comment:
Would it make sense to wrap all this code (maybe except import?) into anonymous function to avoid declaring global name, i.e. eliminate a chance that we'll collide with customer symbols?

I.e. something like:

```
await (async function() {
  /// body of this file goes here;
}) ();
```

(see https://www.javascripttutorial.net/javascript-anonymous-functions/ )

Reply:
I'm not sure if we will be able to test the code if we wrap it in an anonymous function, which was why it is done this way.

Reply:
By looking at test it seems the output assertion is used env vars, so that part is covered, so if there's some way to mock SSM client globally this could be an option.
Other option to solve testing is to test this with live SSM (i.e. not mock SSM and add e2e test for this fragment.

But if that's not very practical then another option is to keep it as is but make sure that function name is long and hard to guess. I.e. `internalAmplifyFunctionBannerResolveSecrets` or something more obfuscated.